typescript/cli/src/deploy/warp.ts

```diff
@@ -313,3 +327,57 @@ async function getWarpCoreConfig(
 
   return warpCoreConfig;
 }
+
+export async function runWarpRouteApply(params: ApplyParams) {
+  const {
+    warpDeployConfig,
+    warpCoreConfig,
+    context: { registry, multiProvider },
+  } = params;
+
+  // Addresses used to get static Ism factories
+  const addresses = await registry.getAddresses();
+
+  // Convert warpCoreConfig.tokens[] into a mapping of { [chainName]: Config }
+  // This allows O(1) reads within the loop
+  const warpCoreByChain = Object.fromEntries(
+    warpCoreConfig.tokens.map((token) => [token.chainName, token]),
+  );
+
+  // Attempt to update Warp Routes
+  // Can update existing or deploy new contracts
+  logGray(`Comparing target and onchain Warp configs`);
+  await promiseObjAll(
+    objMap(warpDeployConfig, async (chain, config) => {
+      try {
+        // Update Warp
+        config.ismFactoryAddresses = addresses[
+          chain
+        ] as ProxyFactoryFactoriesAddresses;
+        const evmERC20WarpModule = new EvmERC20WarpModule(multiProvider, {
+          config,
+          chain,
+          addresses: {
+            deployedTokenRoute: warpCoreByChain[chain].addressOrDenom!,
+          },
+        });
+        const transactions = await evmERC20WarpModule.update(config);
+
+        // Send Txs
+        if (transactions.length) {
+          for (const transaction of transactions) {
+            await multiProvider.sendTransaction(chain, transaction);
+          }
+
+          logGreen(`Warp config updated on ${chain}.`);
+        } else {
+          logGreen(
+            `Warp config on ${chain} is the same as target. No updates needed.`,
+          );
+        }
+      } catch (e) {
+        logRed(`Warp config on ${chain} failed to update.`, e);
+      }
```

Comment:
Are you certain it's better to continue updating the rest if there's a failure instead of stopping and letting the user debug the issue?

Reply:
Not certain. But i'm open to changing this as requirements change. As of now, there are none.

The reason I did this was because we are only doing ism updates, which are mostly isolated for each chain. When updating an ISM we may also deploy one, which is committed right away. Then after deploying, we set a warp route's ISM as a transaction.

If we revert here because an update failed in another chain, we've potentially deployed an ISM for no reason.